extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java

```diff
@@ -217,14 +213,34 @@ void registerProvidersFromAnnotations(CombinedIndexBuildItem indexBuildItem,
             constructor.invokeSpecialMethod(MethodDescriptor.ofConstructor(AnnotationRegisteredProviders.class),
                     constructor.getThis());
 
+            for (AnnotationInstance instance : index.getAnnotations(ResteasyReactiveDotNames.PROVIDER)) {
+                // TODO: we may want to filter out stuff that is server side only
```

Comment:
Here's the part I'm not really sure about.
@geoand do you think registering everything here can have some side effects?

Reply:
You need to honor this: https://quarkus.io/guides/rest-client#rest-client-and-resteasy-interactions (or at least have an equivalent).

Reply:
Exactly. We already do the equivalent on the server side

Reply:
I don't know how I could be unaware of this :) thank you! Fixing